### PR TITLE
added functionality to the homepage

### PR DESCRIPTION
**Pull request by KulakDmitry**

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
```
Comment (alexanderchernetsky) [inline public/index.html]: Remove all unused files from the project

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import { Route, Routes } from "react-router-dom";
+import "./App.css";
+import HomePage from "./components/Homepage/HomePage";
+import MoviePage from "./components/Moviepage/MoviePage";
+
+function App() {
+  const [movieInfo, setMovieInfo] = useState([]);
+
+  return (
+    <div className="App">
```
Comment (alexanderchernetsky) [inline src/App.js]: Do you really need this div wrapper ?

File: src/components/Homepage/HomePage.jsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+export default function HomePage({ setMovieInfo }) {
+  const [movies, setMovies] = useState([]);
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  async function fetchMovieData(id) {
+    const response = await axios.get(
```
Comment (alexanderchernetsky) [inline src/components/Homepage/HomePage.jsx]: You should handle errors in every network request.

File: src/components/Homepage/HomePage.jsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      }
+    );
+    setMovies(response.data.results);
+  }
+
+  return (
+    <>
+      <h1 className="text-4xl mt-10">Most popular movies</h1>
+      <div className="mt-20 grid gap-y-10 grid-cols-5">
+        {movies.map((movie) => (
```
Comment (alexanderchernetsky) [inline src/components/Homepage/HomePage.jsx]: You should add loading state. When the data is loading, the page should NOT be empty. Please add spinner.

File: src/components/Homepage/HomePage.jsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            <img
+              className="rounded"
+              src={`http://image.tmdb.org/t/p/w185/${movie.poster_path}`}
+            />
+            <p className="px-10 py-2">{movie.title}</p>
+          </Link>
+        ))}
+      </div>
+    </>
+  );
```
Comment (alexanderchernetsky) [inline src/components/Homepage/HomePage.jsx:107]: Please add pagination.

**Review by alexanderchernetsky — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 2 above)
Comment (KulakDmitry) [reply src/App.js]: no) fixed

File: public/index.html (hunk 1 above)
Comment (KulakDmitry) [reply public/index.html]: fixed

File: src/components/Homepage/HomePage.jsx (hunk 3 above)
Comment (KulakDmitry) [reply src/components/Homepage/HomePage.jsx]: fixed

File: src/components/Homepage/HomePage.jsx (hunk 4 above)
Comment (KulakDmitry) [reply src/components/Homepage/HomePage.jsx]: fixed

File: src/components/Homepage/HomePage.jsx (hunk 5 above)
Comment (KulakDmitry) [reply src/components/Homepage/HomePage.jsx:107]: fixed

**Comment by alexanderchernetsky** [comment]
@KulakDmitry If you create a PR from homepage branch to develop. You have to fix all comments in the same homepage branch.

**Pull request merged**